src/server/services/procedures/new-york-public-library/new-york-public-library.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+const ApiConsumer = require('../utils/api-consumer');
+const {NewYorkPublicLibraryKey} = require('../utils/api-key');
+const NYPL = new ApiConsumer('NewYorkPublicLibrary', 'http://api.repo.nypl.org/api/v1', {cache: {ttl: 5*60}});
+ApiConsumer.setRequiredApiKey(NYPL, NewYorkPublicLibraryKey);
+
+function listify(item) {
+    return item === undefined ? [] : item instanceof Array ? item : [item];
+}
+
+/**
+ * Search the New York Public Library collection.
```

Comment:
I would add information about paging here. Maybe adding another sentence like: "Search results are stored in pages and the results on the designated page will be returned."

Reply:
This was one of the comments I was referring to before.